File: task3_python/Player.py

import abc
import random
from Pos import Pos


class Player(object):

	__metaclass__ = abc.ABCMeta

	def __init__(self, healthCap, mob,posx,posy,index,game):
		self.MOBILITY = mob
		self.health = healthCap
		self.pos = Pos(posx,posy)
		self.index = index
		self.game = game

	@abc.abstractmethod
	def teleport(self):

		randx = random.randint(0, self.game.Dimension - 1)
		randy = random.randint(0, self.game.Dimension - 1)

		while(self.game.positionOccupied(randx,randy)):
			randx = random.randint(0, self.game.Dimension - 1)
			randy = random.randint(0, self.game.Dimension - 1)


		self.pos.setPos(randx,randy)


	def increaseHealth(self,h):
		self.health += h

	def decreaseHealth(self, h):	# BACK HERE
		self.health -= h
		if(self.health <= 0):
			self.myString = "C"+ self.myString[0]

	@abc.abstractmethod
	def askForMove(self):
		print "Yor health is %d. Your position is (%d,%d). Your mobility is %d."%(self.health, self.pos.x, self.pos.y, self.MOBILITY)

		print"You now have the following options: "
		print"1. Move"
		print"2. Attack"
		print"3. End the turn"

		a = input()

		if a==1:
			print"Specify your target position ( Input 'x y')."
			row,col = raw_input().split()
			row = int(row)
			col = int(col)

			if self.pos.distance1(row,col) > self.MOBILITY :
				print "Beyond your reach. Staying still."
			elif self.game.positionOccupied(row,col):
				print "Position occupied. Cannot move there."
			else:
				self.pos.setPos(row,col)
				self.game.printBoard()
				print"You can now\n1.attacj\n2.End the turn"
				a = input()
				if a ==1 :
					print"Input position to attack. (Input 'x y')"
					attx,atty = raw_input().split()
					attx = int(attx)
					atty = int(atty)
					self.equipment.action(attx,atty)
		elif a==2:
			print"Input position to attack. (Input 'x y')"
			attx,atty = raw_input().split()
			attx = int(attx)
			atty = int(atty)
			self.equipment.action(attx,atty)

		elif a==3:
			return

File: part1/Player.java
import java.util.Random;

public abstract class Player {
	private int MOBILITY;
	protected Pos pos;
	protected int health;
	protected Weapon equipment;
	protected int index;
	protected String myString;
	protected SurvivalGame game;
	
	public Player(int healthCap, int mob, int posx, int posy, int index, SurvivalGame game) {

		this.MOBILITY = mob;
		this.health = healthCap;
		this.pos = new Pos(posx, posy);
		this.index = index;
		this.game = game;
	}

	public Pos getPos() {
		return pos;
	}

	public void teleport() {

		Random rand;
		rand = new Random();
		int randx = rand.nextInt(game.D);
		int randy = rand.nextInt(game.D);
		while (game.positionOccupied(randx, randy)) {
			randx = rand.nextInt(game.D);
			randy = rand.nextInt(game.D);
		}
		pos.setPos(randx, randy);
	}

	public void increaseHealth(int h) {
		this.health += h;
	}

	public void decreaseHealth(int h) {
		this.health -= h;
		if (this.health <= 0)
			this.myString = "C" + this.myString.charAt(0);
	}

	public String getName() {
		return myString;
	}

	public void askForMove() {
		// Print general information
		System.out.println("Your health is " + health
				+ String.format(". Your position is (%d,%d). Your mobility is %d.", pos.getX(), pos.getY(), this.MOBILITY));

		System.out.println("You now have following options: ");
		System.out.println("1. Move");
		System.out.println("2. Attack");
		System.out.println("3. End tne turn");

		int a = SurvivalGame.reader.nextInt();

		if (a == 1) {
			System.out.println("Specify your target position (Input 'x y').");
			int posx = SurvivalGame.reader.nextInt(), posy = SurvivalGame.reader.nextInt();
			if (pos.distance(posx, posy) > this.MOBILITY) {
				System.out.println("Beyond your reach. Staying still.");
			} else if (game.positionOccupied(posx, posy)) {
				System.out.println("Position occupied. Cannot move there.");
			} else {
				this.pos.setPos(posx, posy);
				game.printBoard();
				System.out.println("You can now \n1.attack\n2.End the turn");
				if (SurvivalGame.reader.nextInt() == 1) {
					System.out.println("Input position to attack. (Input 'x y')");
					int attx = SurvivalGame.reader.nextInt(), atty = SurvivalGame.reader.nextInt();
					this.equipment.action(attx, atty);
				}
			}
		} else if (a == 2) {
			System.out.println("Input position to attack.");
			int attx = SurvivalGame.reader.nextInt(), atty = SurvivalGame.reader.nextInt();
			this.equipment.action(attx, atty);
		} else if (a == 3) {
			return;
		}
	}

}
